src/components/frame/Extensions/pages/ExtensionDetailsPage/PermissionsNeededBanner.tsx

```diff
@@ -1,31 +1,74 @@
 import React, { useState } from 'react';
-import { FormattedMessage } from 'react-intl';
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { ColonyRole, Id } from '@colony/colony-js';
+
+import { ActionTypes } from '~redux';
+import { getFormAction } from '~utils/actions';
+import { useAppContext, useAsyncFunction, useColonyContext } from '~hooks';
+import { AnyExtensionData } from '~types';
 import NotificationBanner from '~v5/shared/NotificationBanner';
+import { addressHasRoles } from '~utils/checks';
+
+const displayName =
+  'frame.Extensions.ExtensionDetailsPage.PermissionsNeededBanner';
+
+const MSG = defineMessages({
+  updatedPermission: {
+    id: `${displayName}.updatedPermission`,
+    defaultMessage: 'The required permissions have been updated.',
+  },
+  missingPermission: {
+    id: `${displayName}.missingPermission`,
+    defaultMessage:
+      'This extension is missing some or all of the permissions it needs to work.',
+  },
+  enablePermission: {
+    id: `${displayName}.enablePermission`,
+    defaultMessage: 'Enable permissions',
+  },
+});
 
-const displayName = 'frame.Extensions.PermissionsNeededBanner';
+interface Props {
+  extensionData: AnyExtensionData;
+}
+
+const PermissionsNeededBanner = ({ extensionData }: Props) => {
+  const { colony } = useColonyContext();
+  const { user } = useAppContext();
 
-const PermissionsNeededBanner = () => {
-  // @TODO: Change extension missing permissions functionality
   const [isPermissionEnabled, setIsPermissionEnabled] = useState(false);
+  const errorAction = getFormAction(ActionTypes.EXTENSION_ENABLE, 'ERROR');
+  const successAction = getFormAction(ActionTypes.EXTENSION_ENABLE, 'SUCCESS');
```

Comment:
IIRC, these utils were put in so that components like `ActionForm` and `ActionButton` can get default error and success actions to allow the consumer to omit providing them. Since you have to pass them to `useAsyncFunction` anyway, I don't think there's any benefit in using those over passing `EXTENSION_ENABLE_ERROR` and `EXTENSION_ENABLE_SUCCESS` explicitly.